src/promptflow/promptflow/batch/base_executor_proxy.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import Any, Mapping, Optional
+
+from promptflow.executor._result import AggregationResult, LineResult
+from promptflow.storage._run_storage import AbstractRunStorage
+
+
+class AbstractExecutorProxy:
+    @classmethod
+    def create(
+        cls,
+        flow_file: Path,
+        working_dir: Optional[Path] = None,
+        *,
+        connections: Optional[dict] = None,
+        storage: Optional[AbstractRunStorage] = None
+    ) -> "AbstractExecutorProxy":
+        """Create a new executor"""
+        raise NotImplementedError()
+
+    def destroy(self):
+        """Destroy the executor"""
+        pass
+
+    async def exec_line_async(
```

Comment:
why this is async while exec_aggregation is not?

Reply:
Because `PythonExecutorProxy` implements `exec_aggregation` and it is not an async method. For `exec_line`, it is not used by `PythonExecutorProcy`.

Reply:
Updated.